### ISSUE TITLE
generated images are noisy

### ISSUE THREAD

**Issue by fido20160817** [openai/guided-diffusion#106]
Hi, there is a work based guided-diffusion: https://github.com/WeilunWang/semantic-diffusion-model which implements a semantic diffusion model. But when I try to do sampling, the quality is unsatisfactory. I receive no response by asking its authors, so I have to go to this repository to find answers (reasons may be related to theories of diffusion model).

Here are some generated images on human faces:
https://github.com/WeilunWang/semantic-diffusion-model/issues/14

and more on ade20k and cityscapes:

ddpm sampling
![ADE_val_00000320_use_ddim_False_2](https://github.com/openai/guided-diffusion/assets/21076140/f87d26a3-3d71-4586-acd9-fd651d97f849)  

ddim sampling
![ADE_val_00000320_use_ddim_True_3](https://github.com/openai/guided-diffusion/assets/21076140/b57176fd-ec2f-478b-b7ac-4a269642bd6a) 

ddpm sampling
![frankfurt_000000_000294_leftImg8bituse_ddim_False](https://github.com/openai/guided-diffusion/assets/21076140/c61e56e9-564c-4b88-8e1f-055210f26616) 

ddim sampling
![frankfurt_000000_000294_leftImg8bit](https://github.com/openai/guided-diffusion/assets/21076140/2051071f-81a7-4e7e-83c9-f82660e4c3ed) 

Is there any way to improve the quality of ddim sampling. I tried to set 'guidance scale', but it doesn't help.
I don't know what causes noise in the generated images. Any tips?

**Comment by fido20160817** [comment]
is there any way to improve the quality of ddim sampling. I tried to set 'guidance scale', but it doesn't help.

**Comment by hxy-123-coder** [comment]
Hello！ I want to ask how much the loss of the model you trained converges to?  The images generated by my training are all noise, and I can't see the content of the images at all.

**Comment by kuanglongli** [comment]
Hello! Have you solved this problem? I am in the same situation.

**Comment by hxy-123-coder** [comment]
sorry，but I still struggle with it. if you have some solutions, please tell me



在 2023-07-11 15:36:47，"kuanglongli" ***@***.***> 写道：

你好！ 我想问一下你训练的模型的损失收敛到多少？我训练产生的图像都是噪音，我根本看不到图像的内容。

Hello! Have you solved this problem? I am also trained to generate pictures that are all noise.

**Comment by UESTC-Med424-JYX** [comment]
Could you please solve this problem? I also have this problem.

**Comment by kpms-ignacio-traverso** [comment]
I solved it by reducing the learning rate to 2e-5

**Comment by UESTC-Med424-JYX** [comment]
> I solved it by reducing the learning rate to 2e-5

Thank you, I think this is a good way.

**Comment by yug125lk** [comment]
> Hi, there is a work based guided-diffusion: https://github.com/WeilunWang/semantic-diffusion-model which implements a semantic diffusion model. But when I try to do sampling, the quality is unsatisfactory. I receive no response by asking its authors, so I have to go to this repository to find answers (reasons may be related to theories of diffusion model).
> 
> Here are some generated images on human faces: [WeilunWang/semantic-diffusion-model#14](https://github.com/WeilunWang/semantic-diffusion-model/issues/14)
> 
> and more on ade20k and cityscapes:
> 
> ddpm sampling ![ADE_val_00000320_use_ddim_False_2](https://user-images.githubusercontent.com/21076140/243646771-f87d26a3-3d71-4586-acd9-fd651d97f849.png)
> 
> ddim sampling ![ADE_val_00000320_use_ddim_True_3](https://user-images.githubusercontent.com/21076140/243646237-b57176fd-ec2f-478b-b7ac-4a269642bd6a.png)
> 
> ddpm sampling ![frankfurt_000000_000294_leftImg8bituse_ddim_False](https://user-images.githubusercontent.com/21076140/243646514-c61e56e9-564c-4b88-8e1f-055210f26616.png)
> 
> ddim sampling ![frankfurt_000000_000294_leftImg8bit](https://user-images.githubusercontent.com/21076140/243646375-2051071f-81a7-4e7e-83c9-f82660e4c3ed.png)
> 
> Is there any way to improve the quality of ddim sampling. I tried to set 'guidance scale', but it doesn't help. I don't know what causes noise in the generated images. Any tips?

Hi, could you help me please ,I also tried the code of [https://github.com/WeilunWang/semantic-diffusion-model](url)
I want to train it for image-to-image translation and not for segmentation. How can I set the number of classes and this condition ? : --class_cond True , and thank you.

**Comment by hong-hd** [comment]
> sorry，but I still struggle with it. if you have some solutions, please tell me 在 2023-07-11 15:36:47，"kuanglongli" ***@***.***> 写道： 你好！ 我想问一下你训练的模型的损失收敛到多少？我训练产生的图像都是噪音，我根本看不到图像的内容。 Hello! Have you solved this problem? I am also trained to generate pictures that are all noise. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***>

Hello! Have you solved this problem? I am in the same situation.

**Comment by hong-hd** [comment]
> Hi, there is a work based guided-diffusion: https://github.com/WeilunWang/semantic-diffusion-model which implements a semantic diffusion model. But when I try to do sampling, the quality is unsatisfactory. I receive no response by asking its authors, so I have to go to this repository to find answers (reasons may be related to theories of diffusion model).
> 
> Here are some generated images on human faces: [WeilunWang/semantic-diffusion-model#14](https://github.com/WeilunWang/semantic-diffusion-model/issues/14)
> 
> and more on ade20k and cityscapes:
> 
> ddpm sampling ![ADE_val_00000320_use_ddim_False_2](https://private-user-images.githubusercontent.com/21076140/243646771-f87d26a3-3d71-4586-acd9-fd651d97f849.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMDkxODgsIm5iZiI6MTczMDAwODg4OCwicGF0aCI6Ii8yMTA3NjE0MC8yNDM2NDY3NzEtZjg3ZDI2YTMtM2Q3MS00NTg2LWFjZDktZmQ2NTFkOTdmODQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDA2MDEyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVjMTUwMmZjMDNlYTIyYzNiNGJkZGVhNjNjOTJiOGVlMTQzYWI3YTQ1ZTI3OTU3NWQzZDYzZjliNTA1NDQ2YzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.d62qtYzNbXU5RcVT33krLa-lxLEHSupfc_G-AnyLYDc)
> 
> ddim sampling ![ADE_val_00000320_use_ddim_True_3](https://private-user-images.githubusercontent.com/21076140/243646237-b57176fd-ec2f-478b-b7ac-4a269642bd6a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMDkxODgsIm5iZiI6MTczMDAwODg4OCwicGF0aCI6Ii8yMTA3NjE0MC8yNDM2NDYyMzctYjU3MTc2ZmQtZWMyZi00NzhiLWI3YWMtNGEyNjk2NDJiZDZhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDA2MDEyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ4NjI0ZWIzMjcwMzczMmE2Yzc2OTliNDZkNGU3Zjg0NjlhMjM5MTgxMGMyNzYwNmNkY2M1OGU5ZTY4MWEzODAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jDsz5cMcJvwuLt1xXqkxChkBNZuNcysrTMpdpZBKN_g)
> 
> ddpm sampling ![frankfurt_000000_000294_leftImg8bituse_ddim_False](https://private-user-images.githubusercontent.com/21076140/243646514-c61e56e9-564c-4b88-8e1f-055210f26616.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMDkxODgsIm5iZiI6MTczMDAwODg4OCwicGF0aCI6Ii8yMTA3NjE0MC8yNDM2NDY1MTQtYzYxZTU2ZTktNTY0Yy00Yjg4LThlMWYtMDU1MjEwZjI2NjE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDA2MDEyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4ZTlkMTg1ODNkNWJhZTAzMTdiZDBmOTYwMTlmNDhiMDIzZmIwYTQ1YWE3N2E4YTA0Nzc2NmZiMzU1YTJkOWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ioq6z2IOt8r0aSLsiMSYswhMzlP6L_9YRierOqhSx_4)
> 
> ddim sampling ![frankfurt_000000_000294_leftImg8bit](https://private-user-images.githubusercontent.com/21076140/243646375-2051071f-81a7-4e7e-83c9-f82660e4c3ed.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMDkxODgsIm5iZiI6MTczMDAwODg4OCwicGF0aCI6Ii8yMTA3NjE0MC8yNDM2NDYzNzUtMjA1MTA3MWYtODFhNy00ZTdlLTgzYzktZjgyNjYwZTRjM2VkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDA2MDEyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQwZWFjZmE1YjA5OWJlOTgwYzBmNDg5MWM4MTVkNzM1MWFiOGVlZmZjODFjNmUyMDRiOTZiN2I0MjM1OTJlYzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xrdEBRclxkwfKTTU3F6duSpjHOTr-5OAzWs8K6q6u5E)
> 
> Is there any way to improve the quality of ddim sampling. I tried to set 'guidance scale', but it doesn't help. I don't know what causes noise in the generated images. Any tips?

Hello, I would like to ask how to train rectangular images（H×W）
